tilelang/language/tir/ir.py

```
@@ -10,7 +9,7 @@
 PyPrimExpr = PrimExpr | int | float | bool
 
 
-def serial(start: PrimExpr, stop: PrimExpr = None, *, annotations: dict[str, Any] = None) -> frame.ForFrame:
+def serial(start: int | PrimExpr, stop: int | PrimExpr = None, *, annotations: dict[str, Any] = None) -> Any:
```

Ruff (0.16.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)